SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentFactory.cpp

```diff
@@ -314,9 +316,13 @@ void ComponentFactory::AddProperty(Schema_Object* Object, Schema_FieldId FieldId
 	{
 		// These properties can be set to replicate, but won't serialize across the network.
 	}
+	else if (Property->IsA<UMapProperty>())
+	{
+		UE_LOG(LogComponentFactory, Error, TEXT("Replicated TMaps are not supported."));
+	}
 	else
 	{
-		checkf(false, TEXT("Tried to add unknown property in field %d"), FieldId);
+		UE_LOG(LogComponentFactory, Error, TEXT("Tried to add unknown property in field %d"), FieldId);
```

Comment:
What does native unreal do here? I'm more in favour of explicit check rather than potentially missing a replicated prop.

Reply:
Each property would have its own `NetSerializeItem` which prints an error or does nothing if the property can't be replicated.